src/Microsoft.DotNet.Interactive.Jupyter/JupyterKernelSpecModule.cs

```diff
@@ -36,18 +37,26 @@ public Task<CommandLineResult> InstallKernel(DirectoryInfo sourceDirectory)
 
     public async Task<IReadOnlyDictionary<string, KernelSpec>> ListKernels()
     {
-        var kernelSpecsList = await ExecuteCommand("list", "--json");
-        if (kernelSpecsList.ExitCode == 0)
+        try
         {
-            var results = JsonSerializer.Deserialize<KernelSpecListCommandResults>(string.Join(string.Empty, kernelSpecsList.Output));
-            return results.kernelspecs?.ToDictionary(r => r.Key, r =>
+            var kernelSpecsList = await ExecuteCommand("list", "--json");
+            if (kernelSpecsList.ExitCode == 0)
             {
-                var spec = r.Value?.spec;
-                spec.Name ??= r.Key;
-                return spec;
-            });
+                var results = JsonSerializer.Deserialize<KernelSpecListCommandResults>(string.Join(string.Empty, kernelSpecsList.Output));
+                return results.kernelspecs?.ToDictionary(r => r.Key, r =>
+                {
+                    var spec = r.Value?.spec;
+                    spec.Name ??= r.Key;
+                    return spec;
+                });
+            }
+            else
+            {
+                // fall back to custom lookup logic 
+                return LookupInstalledKernels();
+            }
         }
-        else
+        catch (Exception)
```

Comment:
I am not sure how swallowed exceptions are handled in other places in the code base - but should we log any exceptions that are encountered and swallowed here to a log file here?

Reply:
you are correct. Added logging for this.